### PR TITLE
[JUJU-1639, JUJU-1787] Rewrite api/client/metricsdebug unit tests to use gomock

### PR DESCRIPTION
**Pull request by cderici**
The unit tests for api/client/metricsdebug now use go mock and not juju/testing. All unit tests should pass.

### CODE REVIEW

File: api/client/metricsdebug/client_test.go [hunk 1]
```diff
@@ -7,53 +7,51 @@ import (
...
 	"github.com/juju/juju/rpc/params"
 	"github.com/juju/juju/state"
-	"github.com/juju/juju/testing"
 	"github.com/juju/juju/testing/factory"
 )
 
 type metricsdebugSuiteMock struct {
-	testing.BaseSuite
-	manager *metricsdebug.Client
+	jujutesting.JujuConnSuite
```
Comment (basebandit) [inline api/client/metricsdebug/client_test.go:24]: I would recommend avoiding jujuconnsuite and mock only the required behaviour instead. (*state.State, *state.StatePool & factory.Factory).

**Review by basebandit — APPROVED** [review]
LGTM. QA Passed. Only reservation I have is the introduction of jujuconnsuite.

**Comment by jameinel** [comment]
Thinking about it a bit, and looking at just the diff, you might have actually gotten everything onto gomock and then there actually isn't a need for JujuConnSuite and it was just a step along that path. But we should try to make sure we don't have an accidental dependency there.

**Pull request merged**